lib/Dialect/SV/SVOps.cpp

```diff
@@ -1131,7 +1130,29 @@ LogicalResult IndexedPartSelectInOutOp::inferReturnTypes(
 }
 
 LogicalResult IndexedPartSelectInOutOp::verify() {
-  return verifyIndexedPartSelectOp(*this);
+  unsigned inputWidth = 0, resultWidth = 0;
+  auto opWidth = width();
+
+  if (auto i = input()
+                   .getType()
+                   .cast<InOutType>()
+                   .getElementType()
+                   .dyn_cast<IntegerType>())
+    inputWidth = i.getWidth();
+  else
+    return emitError("input element type must be Integer");
+
+  if (auto resType =
+          getType().cast<InOutType>().getElementType().dyn_cast<IntegerType>())
+    resultWidth = resType.getWidth();
+  else
+    return emitError("result element type must be Integer");
+
+  if (opWidth > inputWidth)
+    return emitError("slice width should not be greater than input width");
+  if (opWidth != resultWidth)
+    return emitError("result width must be equal to slice width");
+  return success();
```

Comment:
I eyeballed this and it looks like the exact same logic, just a nice refactor. 👍

Reply:
I found that not enough code was shared to warrant the dynamic type checks.